### ISSUE TITLE
Increase functionality of `generate_map` to handle outputs

### ISSUE THREAD

**Issue by kvantricht** [WorldCereal/worldcereal-classification#71]
cc @jdegerickx 

`generate_map` currently requires an `output_path` to which the result is automatically downloaded. It would be nice if the method could become somewhat more flexible and also provides more interaction with the actual job running on openEO. Specifally, it could be nice to:

- return the `job_id` (optionally or always?) for interaction with openEO
- return the link to created asset (optionally or always?)
- have the download to local `output_path` functionality optional

But let's have the discussion here on how to make this serve the different needs.

Disclaimer: possibly these requests require first some changes in GFMAP.

**Comment by jdegerickx** [comment]
indeed, depending on where you run this, it might not be necessary to download the actual .tif file, but just receive a link to the job_id and a link to the asset(s) generated so you can view the results in the openeo web editor on CDSE.

For the Maps4GPP project, the generate_map function will also be used to generate WorldCereal maps from within the GPP GeoPortal -> for that purpose, we'll need to build a UDP. This will be covered in a different issue
